recipes/ncbi-cxx-toolkit-public/26.0/conanfile.py

```diff
@@ -0,0 +1,1450 @@
+            if "fastme" in allexports:
+                self.cpp_info.components["fastme"].libs = ["fastme"]
+                self.cpp_info.components["fastme"].requires = ["ORIGLIBS"]
+            if "lapackwrapp" in allexports:
+                self.cpp_info.components["lapackwrapp"].libs = ["lapackwrapp"]
+                self.cpp_info.components["lapackwrapp"].requires = ["LAPACK", "ORIGLIBS"]
+            if "psg_protobuf" in allexports:
+                self.cpp_info.components["psg_protobuf"].libs = ["psg_protobuf"]
+                self.cpp_info.components["psg_protobuf"].requires = ["PROTOBUF", "ORIGLIBS"]
+            if "tables" in allexports:
+                self.cpp_info.components["tables"].libs = ["tables"]
+                self.cpp_info.components["tables"].requires = ["ORIGLIBS"]
+            if "task_server" in allexports:
+                self.cpp_info.components["task_server"].libs = ["task_server"]
+                self.cpp_info.components["task_server"].requires = ["Boost", "ORIGLIBS"]
+            if "tds_ftds100" in allexports:
+                self.cpp_info.components["tds_ftds100"].libs = ["tds_ftds100"]
+                self.cpp_info.components["tds_ftds100"].requires = ["ORIGLIBS"]
+            if "test_dll" in allexports:
+                self.cpp_info.components["test_dll"].libs = ["test_dll"]
+                self.cpp_info.components["test_dll"].requires = ["ORIGLIBS"]
+            if "xncbi" in allexports:
+                self.cpp_info.components["xncbi"].libs = ["xncbi"]
+                self.cpp_info.components["xncbi"].requires = ["ORIGLIBS"]
+#----------------------------------------------------------------------------
+        if self.settings.os == "Windows":
+            self.cpp_info.components["ORIGLIBS"].defines.append("_UNICODE")
+            self.cpp_info.components["ORIGLIBS"].defines.append("_CRT_SECURE_NO_WARNINGS=1")
+        else:
+            self.cpp_info.components["ORIGLIBS"].defines.append("_MT")
+            self.cpp_info.components["ORIGLIBS"].defines.append("_REENTRANT")
+            self.cpp_info.components["ORIGLIBS"].defines.append("_THREAD_SAFE")
+            self.cpp_info.components["ORIGLIBS"].defines.append("_LARGEFILE_SOURCE")
+            self.cpp_info.components["ORIGLIBS"].defines.append("_LARGEFILE64_SOURCE")
+            self.cpp_info.components["ORIGLIBS"].defines.append("_FILE_OFFSET_BITS=64")
+        if self.options.shared:
+            self.cpp_info.components["ORIGLIBS"].defines.append("NCBI_DLL_BUILD")
+        if self.settings.build_type == "Debug":
+            self.cpp_info.components["ORIGLIBS"].defines.append("_DEBUG")
+        else:
+            self.cpp_info.components["ORIGLIBS"].defines.append("NDEBUG")
+        self.cpp_info.components["ORIGLIBS"].builddirs.append("res")
+        self.cpp_info.components["ORIGLIBS"].build_modules = ["res/build-system/cmake/CMake.NCBIpkg.conan.cmake"]
```

Comment:
What is the content of this file? Does it work well with all cmake like generators?

Reply:
it provides with the toolkit cmake extensions, which are useful.

Reply:
_NCBI_begin_app_ and such in test_package are examples of such "cmake extensions"

Reply:
I understand, but CMake modules packaged by libraries may rely on CMake imported targets. Since conan-center recipes don't package CMake imported targets, it can break some CMake module files. Instead conan provides generators:
- `cmake` or the new `CMakeToolchain` (the glue bewteen conan and downtream CMake files)
- `cmake_find_package`/`cmake_find_package_multi` or the new `CMakeDeps` (emulate CMake Find/config files)

`build_modules` are files automatically included in generators. So it means that downstream doesn't even need to include these files themself. Since I don't see any CMake config file in NCBI C++ toolkit, it's probably expected to consumers that they manually include this file instead, isn't it?

Reply:
these "extensions" are optional, not mandatory. this is only a convenience.